apps/wallet-connect/src/hooks/useWalletConnect.tsx

```diff
@@ -48,22 +49,75 @@ const useWalletConnect = () => {
         setWcClientData(payload.params[0].peerMeta);
       });
 
-      wcConnector.on('call_request', (error, payload) => {
+      wcConnector.on('call_request', async (error, payload) => {
         if (error) {
           throw error;
         }
 
+        if (payload.method === 'eth_sign') {
+          const [address, messageHash] = payload.params;
+
+          if (address !== safe.safeAddress || !messageHash.startsWith('0x')) {
+            wcConnector.rejectRequest({
+              id: payload.id,
+              error: {
+                message: 'The address or message hash is invalid',
+              },
+            });
+          }
+
+          const callData = encodeSignMessageCall(messageHash);
+          try {
+            await sdk.txs.send({
+              txs: [
+                {
+                  to: safe.safeAddress,
+                  value: '0x0',
+                  data: callData,
+                },
+              ],
+            });
+
+            wcConnector.approveRequest({
+              id: payload.id,
+              result: '',
```

Comment:
maybe we should return `0x` 🤔 on low level ethereum representation that is more correct